feat: deprecating carousel slides

mcbridemarci opened the pull request:
<!--
Please submit this PR as a draft initially.
Do not mark it as "Ready for review" until the template has been completely filled out, and PR status checks have passed at least once.
-->

## **Description**

This PR focuses on the Deprecation of the following carousel slides: 
        - [x] BACKUPANDSYNC_SLIDE
        - [x] SWEEPSTAKES_SLIDE
        - [x] MULTI_SRP_SLIDE
        - [x] SMART_ACCOUNT_UPGRADE_SLIDE
        - [x] BRIDGE_SLIDE
        - [x] REMOTE_MODE_SLIDE
        - [x] CASH_SLIDE 
        - [x] basic_functionality
Migrating the following carousel slides:
        - [x] FUND_SLIDE
        - [x] SOLANA_SLIDE
        - [x] CARD_SLIDE
        - [x] downloadMobileApp


This PR also introduces Specific card placement via contentful. Which allows the user to specify what slot should a card specify.

Additional information can be found here:
https://docs.google.com/document/d/1qfntKBDNQVCQzbBNcd0Qb8rzsVn8PPUIjaKPk5R9dBY/edit?tab=t.0

<!--
Write a short description of the changes included in this pull request, also include relevant motivation and context. Have in mind the following questions:
1. What is the reason for the change?
2. What is the improvement/solution?
-->

[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/35109?quickstart=1)

## **Changelog**

<!--
If this PR is not End-User-Facing and should not show up in the CHANGELOG, you can choose to either:
1. Write `CHANGELOG entry: null`
2. Label with `no-changelog`

If this PR is End-User-Facing, please write a short User-Facing description in the past tense like:
`CHANGELOG entry: Added a new tab for users to see their NFTs`
`CHANGELOG entry: Fixed a bug that was causing some NFTs to flicker`

(This helps the Release Engineer do their job more quickly and accurately)
-->

CHANGELOG entry:

## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**

<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [x] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [x] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [x] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

**Priority Placement and card placement is respected:** 
<img width="929" height="373" alt="Screenshot 2025-08-19 at 2 33 32 PM" src="https://github.com/user-attachments/assets/6ce33fe1-5801-4a69-8fc6-b384d7e881f3" />

<img width="1131" height="274" alt="Screenshot 2025-08-19 at 2 34 11 PM" src="https://github.com/user-attachments/assets/c982637a-acb7-4a56-91bf-8a5e63fc6297" />

<img width="1271" height="637" alt="Screenshot 2025-08-19 at 2 37 09 PM" src="https://github.com/user-attachments/assets/a95cca3c-5542-4650-9b7e-98c107b62ad8" />

Prithpal-Sooriya reviewed (dismissed):
LGTM, E2E tests may need some tweaks to get CI to pass.

mcbridemarci committed:
feat: deprecated SWEEPSTAKES_SLIDE, MULTI_SRP_SLIDE, SMART_ACCOUNT_UP, GRADE_SLIDE, BRIDGE_SLIDE, REMOTE_MODE_SLIDE, CASH_SLIDE, BASIC_FUNCTIONALITY

mcbridemarci committed:
fix: carousel test

Prithpal-Sooriya committed:
Merge branch 'main' of github.com:MetaMask/metamask-extension into feat/carousel_migration

Prithpal-Sooriya committed:
refactor: update locales

Prithpal-Sooriya committed:
test: fix unit tests

Prithpal-Sooriya committed:
test: fix onboarding e2e tests with comments

we need to decide if we want to show or hide the basic functionality banner now that we are fully fetching carousels remotely

Prithpal-Sooriya commented on test/e2e/tests/onboarding/onboarding.spec.ts:
```diff
@@ -279,8 +279,10 @@ describe('MetaMask onboarding', function () {
         const homePage = new HomePage(driver);
         await homePage.checkPageIsLoaded();
+        // NOTE - we do not show a carousel modal when basic functionality if off
+        // TODO - decide if we want to keep or remove this behaviour
         // check the basic functionality is off warning message is displayed
-        await homePage.checkBasicFunctionalityOffWarnigMessageIsDisplayed();
+        // await homePage.checkBasicFunctionalityOffWarnigMessageIsDisplayed();
```
Keeping this in for now, we can remove in a fast follow PR or add back this check (which requires us to produce "offline-carousels"

mcbridemarci committed:
removed basic functionality

mcbridemarci committed:
Merge branch 'main' into feat/carousel_migration

mcbridemarci commented on ui/hooks/useCarouselManagement/useCarouselManagement.ts:
```diff
@@ -63,12 +37,62 @@ export function isActive(
+  }
+
+  let up = 0;
+  for (let i = 0; i < placed.length && up < unplaced.length; i++) {
```
We've set the max limit and uniqueness on the Contentful side.

seaona commented on test/e2e/tests/carousel/carousel.spec.ts:
```diff
@@ -6,151 +6,167 @@ import { MAX_SLIDES } from '../../../../ui/components/multichain/carousel/consta
+        );
 
-        const slideIds = SLIDE_IDS;
+        const visibleCount = Math.min(slides.length, MAX_VISIBLE_SLIDES);
```
hi there :wave: I'm reviewing now the e2e changes from this PR and we are never going to get any slides, because we mock any live API request, meaning the Launchdarkly will never return Contentful slides, unless we mock them.

I would suggest, instead of having this `if (!hasCarousel)` conditional, which will bypass the tests, we could instead, test for specific content, by passing a custom mock.

This will look something like this:


```
const FEATURE_FLAGS_URL = 'https://client-config.api.cx.metamask.io/v1/flags';

export const mockContentfulFeatureFlag = (mockServer: Mockttp) =>
  mockServer
    .forGet(FEATURE_FLAGS_URL)
    .withQuery({
      client: 'extension',
      distribution: 'main',
      environment: 'dev',
    })
    .thenCallback(() => {
      return {
        statusCode: 200,
        json: {} // whatever slides/content you want to pass, following the correct data format
      };
    });
```

and then you should pass this into the test setup, where you want that to apply:

```
await withFixtures(
      {
        fixtures: new FixtureBuilder().build(),
        title: this.test?.fullTitle(),
        ....
        testSpecificMock: mockContentfulFeatureFlag, /// this right here
      },
```

what do you think? Does it make sense to create a follow up PR with this change?

Thank you in advance :pray:
